rust/Cargo.lock

```diff
@@ -1622,53 +1987,78 @@ dependencies = [
 ]
 
 [[package]]
-name = "zcash_note_encryption"
-version = "0.0.0"
-source = "git+https://github.com/zcash/librustzcash.git?rev=0a1ed9b8acf336bdc20ba49083c66f114ecf6877#0a1ed9b8acf336bdc20ba49083c66f114ecf6877"
+name = "zcash_encoding"
+version = "0.1.0"
+source = "git+https://github.com/zcash/librustzcash.git?rev=37fc28634e811fdf8db9274a0080cff17c6b6a09#37fc28634e811fdf8db9274a0080cff17c6b6a09"
 dependencies = [
- "blake2b_simd",
  "byteorder",
- "crypto_api_chachapoly",
- "ff",
- "group",
- "rand_core 0.5.1",
+ "nonempty",
+]
+
+[[package]]
+name = "zcash_note_encryption"
+version = "0.1.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "33f84ae538f05a8ac74c82527f06b77045ed9553a0871d9db036166a4c344e3a"
+dependencies = [
+ "chacha20",
+ "chacha20poly1305",
+ "rand_core",
+ "subtle",
+]
+
+[[package]]
+name = "zcash_note_encryption"
+version = "0.1.0"
+source = "git+https://github.com/zcash/librustzcash.git?rev=37fc28634e811fdf8db9274a0080cff17c6b6a09#37fc28634e811fdf8db9274a0080cff17c6b6a09"
+dependencies = [
+ "chacha20",
+ "chacha20poly1305",
+ "rand_core",
  "subtle",
 ]
```

Comment:
This duplication seems undesirable.

Reply:
That's weird, I wonder how that got introduced.

Reply:
Oh, this is just because of the patch dependency.